src/barplot.ts

```diff
@@ -69,41 +78,61 @@ export class BarPlot extends AVisInstance implements IVisInstance {
     return <Element>this.$node.node();
   }
 
+  transform(scale?: [number, number], rotate: number = 0) {
+    const bak = {
+      scale: this.options.scale || [1, 1],
```

Comment:
If you want to backup an array, you need to clone it using `array.slice(0)`, because the data inside the array might change.